packages/reporter/src/reporters/gitlab/builders/CodeQualityReportBuilder.ts

```diff
@@ -0,0 +1,91 @@
+import type {
+  CodeQualityReport,
+  CodeQualityIssue,
+  CodeQualitySeverity
+} from '../types';
+import type { Issue } from '@sectester/scan';
+import crypto from 'node:crypto';
+
+/**
+ * Builder for creating GitLab Code Quality reports from SecTester scan issues.
+ * Converts multiple issues into the GitLab Code Quality format.
+ */
+export class CodeQualityReportBuilder {
+  constructor(
+    private readonly issues: Issue[],
+    private readonly testFilePath: string
+  ) {}
+
+  /**
+   * Builds a GitLab Code Quality report from the provided issues.
+   *
+   * @returns An array of Code Quality issues in GitLab format
+   */
+  public build(): CodeQualityReport {
+    return this.issues.map(issue => this.convertIssueToCodeQualityIssue(issue));
+  }
+
+  /**
+   * Converts a SecTester issue to GitLab Code Quality issue format.
+   *
+   * @param issue - The SecTester issue to convert
+   * @returns The corresponding GitLab Code Quality issue
+   */
+  private convertIssueToCodeQualityIssue(issue: Issue): CodeQualityIssue {
+    const { originalRequest, name, severity } = issue;
+    const description = `${name} vulnerability found at ${originalRequest.method.toUpperCase()} ${originalRequest.url}`;
+
+    const fingerprint = this.createFingerprint(issue);
+
+    // Map sectester severity to GitLab severity
+    const gitlabSeverity = this.mapSeverity(severity);
+
+    return {
+      description,
+      check_name: name,
+      fingerprint,
```

Comment:
Shorthand field first

```suggestion
      description,
      fingerprint,
      check_name: name,
```